perun-dispatcher/src/main/java/cz/metacentrum/perun/dispatcher/scheduling/SchedulingPool.java

```diff
@@ -38,10 +38,15 @@
 	int addToPool(Task task, DispatcherQueue dispatcherQueue)
 			throws InternalErrorException;
 
+	void addTaskSchedule(Task task, int delayCount);
```

Comment:
Hi @dsarman, I know this class is poorly documented, but can you please add javadoc for some methods. I'd like to know the difference between addTaskSchedule() and addToPool() and how they work together. Because I see, that you call both, when new Task is created (comes from audit message) and also in some other places.
You are not using it in ``reloadTasks()`` which is called when dispatcher is restarted. Then Tasks from DB are directly inserted into pool (not using addToPool()) and no TaskSchedule is created for them (context is lost in this case). Is this OK ? 
I believe not, since ``TaskSchedulerImpl`` is now managing propagation (checking pool/queue for Tasks and plan them) and it queries for ``TaskSchedule`` by ``getWaitingTasksQueue()``. But such Tasks have no TaskSchedule.

Can you please fix it or better explain it, maybe it works differently than I understand it. Thank you.